### ISSUE TITLE
YSA LSA script - standard deviation too high for beta distribution 

### ISSUE THREAD

**Issue by Ellie7** [Ellie7/parrot-proj#25]
getting the standard deviation error message again when making the sensitivity to 10% increase in pi plot (in YSA LSA script) 
- lines 225-226 create the data frames with the adjusted survival values for juvenile and adult stages 
- lines 241-242 create the matrices by applying ysaFunc to the data frames

**Comment by tdjames1** [comment]
Aha, the problem is that when you change the survival value `p` then it changes the value of `p(1 − p)` which is the upper bound for the variance allowed for the beta distribution.

So, you will need to change `ysaFunc` to adjust the SD value, after all.

Here's a replacement for lines 16-21 in `ysa functions.R` which uses `purrr` to get the beta values, checking if the SD value is OK and replacing it if not.

```
  #ps
  p <- purrr::map2(dataSource$pi, dataSource$piSD, function(mu, sdev) {
    ## check if variance < (1-p) * p
    if (sdev^2 < (1 - mu)*mu) {
      ## OK to use sdev in betaval function
      betaval(mu, sdev, fx=runif(1))
    } else {
      ## Replace sdev with allowable value
      betaval(mu, sqrt((1 - mu)*mu) - 0.01, fx=runif(1))      
    }
  })
  names(p) <- c("p1a", "p1b", "p1c", "p2", "p3")
  ## this adds elements of the list to the current environment
  list2env(p, envir = environment())
```

**Comment by Ellie7** [comment]
yeah thats what I thouhg the problem was! thank you! 
 should I change the yellow data frame back to the original now? or keep that fix?

**Comment by tdjames1** [comment]
Yes, you can go back to setting the SD to sqrt(0.1722) or whatever the value for the variance of the imputed value was.

**Comment by Ellie7** [comment]
will do :) thank you so much. 

fixed with commit 8e17cb7
